client/templates/userprofile/settings.js

```diff
@@ -115,13 +157,17 @@ Template.profileSettings.events({
             }
 
         }
+        }
 
             //Session.set("errors", "");
-            Meteor.call("User.update", Meteor.userId(),"realname", realname);
+           // Meteor.call("User.update", Meteor.userId(),"realname", realname);
 
+        setTimeout(function() {
```

Comment:
Please use SimpleSchema and Autoform for validations. setTimeout may introduce bugs and unnecessary delays. Rely on reactive data sources (reactive variables, etc) for UI updates.